### PR TITLE
M3-4900 Add: Create new entity transfer (Skeleton)

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Putting this up to get some feedback on the general approach before I get too far down the rabbit hole.

## FAQ

Q: Does this match the designs?
A: No

Q: Does it include all the important functionality?
A: lol no

Q: Does it use real data?
A: also no

Q: Is it hideous?
A: God yes

Q: Is the code clean and tested at least?
A: Stop hounding me

Q: What does it do, exactly?
A: You should be able to select all/some entities, deselect them, and submit the form to generate a payload.

## Questions for reviewers

- Is the abstraction for the tables worth it? I'm hoping to simplify the process for adding other entity types down the road but the savings is kind of marginal.
- What about the table row with a checkbox? Same reason, we want to do bulk actions everywhere eventually so I thought it would be helpful to have, but the actual savings are on the border between worth it/not worth it.

### CODE REVIEW

File: packages/manager/src/features/EntityTransfers/EntityTransfersCreate/transferReducer.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import produce from 'immer';
```
Comment (Jskobos) [inline packages/manager/src/features/EntityTransfers/EntityTransfersCreate/transferReducer.ts:1]: Of course there's a reducer!

**Review by johnwcallahan — APPROVED** [review]
Way to go for putting this together so quickly; this works exactly how I'd expect and the architecture makes sense to me. 

> Is the abstraction for the tables worth it?

Yes, especially once we add styling.

> What about the table row with a checkbox?

Yes, same reason. Depending on how specific it gets styled, I might consider putting the component in the EntityTransfers feature directory instead of the components directory, but that could be debated either way.

Reducer looks good.

**Review by tiffwong — APPROVED** [review]
Woah, great job getting this set up!

**Review by DevDW — APPROVED** [review]
This is looking good; I think the infrastructure for the down-the-road pieces will prove useful 👍

**Pull request merged**